index.js

```diff
@@ -1,21 +1,30 @@
-import isUndefined from 'lodash/lang/isUndefined';
+'use strict';
+
+var isUndefined = require('lodash.isundefined');
 
 /**
  * Recursive dispatch, returns a function which iterates a series of commands
  * looking for one to handle the target and return a value. The commands adhere
  * to an interface of (target, fn) where the returned dispatch fn is passed
  * along to each command, where it can be used by the command. If the target
- * cannot be handled by a command the command returns undefined.
+ * cannot be handled by a command the command returns undefined. If none of the
+ * supplied commands handle the target, an error is thrown.
  *
  * @param commands
  * @returns {Function}
  */
-function dispatch(...commands) {
+
+module.exports = function dispatch() {
+  var commands = new Array(arguments.length);
+  for(var i = 0; i < commands.length; i += 1) {
+    commands[i] = arguments[i];
+  }
+
   return function fn(target) {
-    let result;
+    var result;
 
-    for (let command of commands) {
-      result = command(target, fn);
+    for (var j = 0; j < commands.length; j += 1) {
+      result = commands[j](target, fn);
 
       if (!isUndefined(result)) {
```

Comment:
What do you think about getting rid of the lodash dependency entirely by using `if (result !== undefined)`?